services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -243,6 +243,17 @@ async function handleFailedWorkItems(
             jobMessage = `Maximum allowed errors ${env.maxErrorsForJob} exceeded. See the errors fields for more details`;
             logger.warn(jobMessage);
             continueProcessing = false;
+          } else {
+            const successCount = await workItemCountWithStatusForJobID(tx, job.jobID, WorkItemStatus.SUCCESSFUL);
```

Comment:
The checking logic seems off. I think the use case of this ticket to guard against harmony blindly pushing all the work to a failed service. We don't want to check all work items within a job, but only work items on the current processing workflow step. Imagine a 5 step request with 10k granules, the first 4 steps are all successful and the last step all failed. I don't this checking logic will handle the scenario.